### DISCUSSION TITLE
Render AePPL’s model using `daft`

### DISCUSSION THREAD

**Discussion by rlouf** [aesara-devs/aeppl#189] [Ideas]
[daft](https://docs.daft-pgm.org/en/latest/) is a really neat Python library to plot probabilistic graphical models. Using this to render probabilistic models built with Aesara would be a good project for someone who wants to familiarise themselves with Aesara/AePPL.

The only caveat is that the library does not position the nodes automatically so you would have to write some code to do that. I've done this kind of thing in the past, it's not easy but you can find some CS papers on the topic.

**Comment by ricardoV94** [comment]
Why not use something like graphviz which takes care of the spatial layout for you? https://graphviz.readthedocs.io/en/stable/examples.html

**Reply by rlouf** [reply]
That's a good idea! Daft makes really good plots though, and we could use the same algorithm as graphviz's. Graphviz needs some tweaking anyway, see https://github.com/pyro-ppl/pyro/issues/2980
